servant-docs/src/Servant/Docs/Internal.hs

```diff
@@ -931,6 +930,17 @@ instance HasDocs Raw where
   docsFor _proxy (endpoint, action) _ =
     single endpoint action
 
+instance (KnownSymbol summ, KnownSymbol desc, HasDocs api)
+  => HasDocs (Summary summ :> Description desc :> api) where
```

Comment:
Write a test. I'm surprised if we don't get into problems with overlapping instances.

Reply:
What about `Description desc :> Summary sum :> api`.

I think saner approach is to change `Description` and `Summary` instances so they can find a desc/sum note, and alter it.